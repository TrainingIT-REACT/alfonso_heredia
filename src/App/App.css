.App {
  margin: auto;
  max-width: 90%;
}

.App h1 {
  text-align: center;
}
