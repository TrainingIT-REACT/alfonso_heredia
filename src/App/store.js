import { createStore, combineReducers, applyMiddleware } from "redux";
import promise from 'redux-promise-middleware';

// Reducers
import albums from './Reducers/albums';
import songs from './Reducers/songs';


export default createStore(
  combineReducers({albums, songs}),
  applyMiddleware(promise())
);
