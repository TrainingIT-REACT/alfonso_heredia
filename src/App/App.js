import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import { Provider } from "react-redux";

import Home from './Components/Home/Home';
import Albums from './Components/Albums/Albums';
import About from './Components/About/About';
import Songs from './Components/Songs/Songs';
import NavBar from './Components/NavBar/NavBar';
import PrivateRoute from './Components/PrivateRoute/PrivateRoute';
import Login from './Components/Login/Login';
import Profile from './Components/Profile/Profile';

import Error from "./Components/Error/Error";

import UserContext from './Contexts/UserContext'

// Store
import store from './store';

// Css
import './bootstrap.min.css';
import './App.css';

const NotFound = () => <p>Ups! Parece que aqu√≠ no hay nada (404)</p>;


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      signedIn: false,
      username: '',
      updateUser: this.updateUser
    }
  }

  onReset = () => this.setState({});
  updateUser = (signedIn, username) => {
    this.setState(() => ({ signedIn, username }));
  }

  render() {
    return (
      <Provider store={store}>
        <div className="App">
          <Error onReset={this.onReset} message="Ops! Algo ha salido mal">
            <Router>
              <UserContext.Provider value={this.state}>
                <NavBar />
                <Switch>
                  <Route path="/" exact component={Home} />
                  <Route path="/albumes" exact component={Albums} />
                  <Route path="/albumes/:id" component={Songs} />
                  <Route path="/about" component={About} />
                  <Route path="/login" component={Login} />
                  <PrivateRoute path="/profile" component={Profile} />
                  <Route component={NotFound} />
                </Switch>
              </UserContext.Provider>
            </Router>
          </Error>
        </div>
      </Provider>
    );
  }
}

export default App;
