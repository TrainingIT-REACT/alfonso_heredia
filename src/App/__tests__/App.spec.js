import React from 'react';
import { render } from 'enzyme';

// Componente
import App from '../App';

describe(App, () => {
  it('renders correctly', () => {
    const wrapper = render(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  describe("Render", () => {
    let wrapper;

    beforeEach(() => {
      wrapper = render(<App />);
    });

    it('should add the HTML elements', () => {
      expect(wrapper.is('div')).toBeTruthy();
      expect(wrapper.find('h1').length).toBe(1);
      expect(wrapper.find('h2').length).toBe(2);
    });
  });
})
