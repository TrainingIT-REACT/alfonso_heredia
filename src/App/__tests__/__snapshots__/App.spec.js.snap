// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  class="App"
>
  <nav
    class="navbar navbar navbar-dark bg-dark"
  >
    <ul
      class="nav"
    >
      <li
        class="nav-item"
      >
        <a
          aria-current="page"
          class="nav-link active"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/albumes"
        >
          Albumes
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/about"
        >
          About
        </a>
      </li>
      <li
        class="nav-item"
      >
        <a
          class="nav-link"
          href="/login"
        >
          Login
        </a>
      </li>
    </ul>
    <form
      class="form-inline"
    >
      <input
        aria-label="Search"
        class="form-control mr-sm-2"
        placeholder="Search"
        type="search"
      />
      <button
        class="btn btn-outline-success my-2 my-sm-0"
        type="submit"
      >
        Search
      </button>
    </form>
  </nav>
  <div
    class="page"
  >
    <h1
      class="text-center"
    >
      Reactify
    </h1>
    <h2
      class="bg-secondary text-white text-center rounded m-3"
    >
      Canciones mas escuchadas
    </h2>
    <div
      class="songs-wrapper"
    >
      <div
        class="songs"
      />
    </div>
    <div
      class="clearfix"
    />
    <h2
      class="bg-secondary text-white text-center rounded m-3"
    >
      Albumes mas descargados
    </h2>
    <div />
  </div>
</div>
`;
